application/client/src/app/ui/views/workspace/headers/component.ts

```diff
@@ -63,13 +68,33 @@ export class ColumnsHeaders extends ChangesDetector implements AfterContentInit
             this.session.stream.subjects.get().rank.subscribe(() => {
                 this.markChangesForCheck();
             }),
+            this.controller.subjects.get().visibility.subscribe(() => {
+                this.headers = this.controller.headers
+                .filter((h) => h.visible)
+                .map((h) => new RenderedHeader(h));
+                this.markChangesForCheck();
+            })
         );
         this.headers = this.controller.headers
             .filter((h) => h.visible)
             .map((h) => new RenderedHeader(h));
```

Comment:
I think here can be a problem. Property `headers` in `html` uses index of header to update its properties (`visibility` and `width`), but `index` can be changed... for example, you have at initianal state columns
```
column - index
a - 0
b - 1
c - 2
d - 3
e - 4
```
At some moment user hides column `b` and you will get
```
column - index
a - 0
c - 1
d - 2
e - 3
```
Indexes of columns c, d, and e are now changed...it's different in view and controller